src/spec_swagger/swagger2/core.cljc

```diff
@@ -1,6 +1,87 @@
 (ns spec-swagger.swagger2.core
   (:require [clojure.walk :as walk]
-            [spec-swagger.swagger2.schema :as schema]))
+            [clojure.spec.alpha :as s]
+            [spec-tools.json-schema :as json-schema]
+            [spec-tools.visitor :as visitor]
+            [spec-tools.core :as st]))
+
+;;
+;; conversion
+;;
+
+(defn- spec-dispatch [dispatch _ _ _] dispatch)
+(defmulti accept-spec spec-dispatch :default ::default)
+
+(defmethod accept-spec 'clojure.core/float? [_ _ _ _] {:type "number" :format "float"})
+(defmethod accept-spec 'clojure.core/double? [_ _ _ _] {:type "number" :format "double"})
+(defmethod accept-spec 'clojure.core/nil? [_ _ _ _] {})
+
+;; anyOf is not supported
+(defmethod accept-spec 'clojure.spec.alpha/or [_ _ children _]
+  (assoc
+    (first children)
+    :x-anyOf children))
+
+;; allOf is not supported
+(defmethod accept-spec 'clojure.spec.alpha/and [_ _ children _]
+  (assoc
+    (first children)
+    :x-allOf children))
+
+;; anyOf is not supported
+(defmethod accept-spec 'clojure.spec.alpha/alt [_ _ children _]
+  (assoc
+    (first children)
+    :x-anyOf children))
+
+;; anyOf is not supported
+(defmethod accept-spec 'clojure.spec.alpha/cat [_ _ children _]
+  {:type "array"
+   :items (assoc
+            (first children)
+            :x-anyOf children)})
+
+;; heterogeneous lists not supported
+(defmethod accept-spec 'clojure.spec.alpha/tuple [_ _ children _]
+  {:type "array"
+   :items {}
```

Comment:
I believe that the generated schemas are all swagger2 compatible and all generated schemas are now validated with ring-swagger validator. Or did I miss something?

Another story is that not all generated schemas are not fully accurate as they present only a subset on available values.